eventuate-tram-cdc-connector/src/main/java/io/eventuate/tram/cdc/connector/BinlogEntryToMessageConverter.java

```diff
@@ -1,17 +1,34 @@
 package io.eventuate.tram.cdc.connector;
 
+import io.eventuate.common.id.IdGenerator;
 import io.eventuate.common.json.mapper.JSonMapper;
 import io.eventuate.local.common.BinlogEntry;
 import io.eventuate.local.common.BinlogEntryToEventConverter;
 
+import java.util.HashMap;
 import java.util.Map;
 
 public class BinlogEntryToMessageConverter implements BinlogEntryToEventConverter<MessageWithDestination> {
+
+  public IdGenerator idGenerator;
+
+  public BinlogEntryToMessageConverter(IdGenerator idGenerator) {
+    this.idGenerator = idGenerator;
+  }
+
   @Override
   public MessageWithDestination convert(BinlogEntry binlogEntry) {
+
+    Map<String, String> headers = JSonMapper.fromJson(binlogEntry.getJsonColumn("headers"), Map.class);
+
+    if (!headers.containsKey("ID")) {
+      headers = new HashMap<>(headers);
+      headers.put("ID", idGenerator.genId(binlogEntry.getLongColumn("xid")).asString());
```

Comment:
"xid" should not be a magic value. There should be a constant defined in eventuate-common

Reply:
Agree, but usually we don't use constants for column names, and having only one can look a little inconsistent, no?
Or should we think about constants for all columns?